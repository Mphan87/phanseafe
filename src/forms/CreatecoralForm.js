import React, { useState } from "react";
import { useNavigate } from "react-router-dom";

function CreatecoralForm({ createcoral }) {
  const navigate = useNavigate();
  const [formData, setFormData] = useState({
    handle: "",
    name: "",
    image: "",
    description: "",
    type: "",
  });

  const [formErrors, setFormErrors] = useState([]);
  async function handleSubmit(evt) {
    evt.preventDefault();
    let result = await createcoral(formData);
    console.log(result)
    if (result.success) {
      console.log(result.success)
      navigate("/coral");
    } else {
      setFormErrors(result.errors);
    }
  }

  function handleChange(evt) {
    const { name, value } = evt.target;
    setFormData(data => ({ ...data, [name]: value }));
    console.log(setFormData)
  }

  return (
    <div>
    <h3 className="fs-title">Sign Up</h3>
    <form id="msform" onSubmit={handleSubmit}>

        <div className="form-group">
          <label>Handle:</label>
          <input
            name="handle"
            className="form-control"
            value={formData.handle.replace(/ /g, "-")}
            onChange={handleChange}
          />
        </div>

        <div className="form-group">
          <label>Name:</label>
          <input
            name="name"
            className="form-control"
            value={formData.name}
            onChange={handleChange}
          />
        </div>

        <div className="form-group">
          <label>Image:</label>
          <input
            name="image"
            className="form-control"
            value={formData.image}
            onChange={handleChange}
          />
        </div>

        <div className="form-group">
          <label>Description:</label>
          <input
            name="description"
            className="form-control"
            value={formData.description}
            onChange={handleChange}
          />
        </div>
        
        <div className="form-group">
          <form id="dropdown">
          <label>Type:</label>
            <select name="type" onChange={handleChange}>
              <option value="LPS">LPS</option>
              <option value="SPS">SPS</option>
              <option value="Zoanthids">Zoanthids</option>
              <option value="Mushrooms">Mushrooms</option>
              <option value="Anemones">Anemones</option>
              <option value="Soft">Soft</option>
            </select>
          </form>
        </div>

        <button
          type="submit"
          className="next action-button"
          onSubmit={handleSubmit}>
          Submit
        </button>
      </form>
    </div>
  );
}

export default CreatecoralForm;