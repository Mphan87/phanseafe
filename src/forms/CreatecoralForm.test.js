import React from "react";
import { render } from "@testing-library/react";
import CreatecoralForm from "./CreatecoralForm";
import { MemoryRouter } from "react-router";

it("matches snapshot", function () {
  const { asFragment } = render(
      <MemoryRouter>
        <CreatecoralForm />
      </MemoryRouter>,
  );
  expect(asFragment()).toMatchSnapshot();
});
