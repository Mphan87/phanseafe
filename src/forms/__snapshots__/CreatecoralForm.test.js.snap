// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <h3
      class="fs-title"
    >
      Sign Up
    </h3>
    <form
      id="msform"
    >
      <div
        class="form-group"
      >
        <label>
          Handle:
        </label>
        <input
          class="form-control"
          name="handle"
          value=""
        />
      </div>
      <div
        class="form-group"
      >
        <label>
          Name:
        </label>
        <input
          class="form-control"
          name="name"
          value=""
        />
      </div>
      <div
        class="form-group"
      >
        <label>
          Image:
        </label>
        <input
          class="form-control"
          name="image"
          value=""
        />
      </div>
      <div
        class="form-group"
      >
        <label>
          Description:
        </label>
        <input
          class="form-control"
          name="description"
          value=""
        />
      </div>
      <div
        class="form-group"
      >
        <label>
          Type:
        </label>
        <select
          name="type"
        >
          <option
            value="LPS"
          >
            LPS
          </option>
          <option
            value="SPS"
          >
            SPS
          </option>
          <option
            value="Zoanthids"
          >
            Zoanthids
          </option>
          <option
            value="Mushrooms"
          >
            Mushrooms
          </option>
          <option
            value="Anemones"
          >
            Anemones
          </option>
          <option
            value="Soft"
          >
            Soft
          </option>
        </select>
      </div>
    </form>
    <button
      class="next action-button"
      type="submit"
    >
      Submit
    </button>
  </div>
</DocumentFragment>
`;
